arcgis-ios-sdk-samples/Maps/Create and save a map/CreateSaveMapViewController.swift

```diff
@@ -47,6 +48,9 @@ class CreateSaveMapViewController: UIViewController, CreateOptionsViewController
             "SaveAsViewController"
         ]
         
+        apiKey = AGSArcGISRuntimeEnvironment.apiKey
+        AGSArcGISRuntimeEnvironment.apiKey = ""
```

Comment:
Is clearing the environment API key really necessary? The key set on the basemap should override the environment key anyway, right?

Reply:
To my understanding: it is to prevent the map getting saved to the account which created the API key.

According to Trevor: the expected behavior with API key being set on `AGSArcGISRuntimeEnvironment` is that the `Create and save map` sample will save the map to the key-issuing account (runtime SDK account) without asking for OAuth credentials. But that is not what we are seeing, and we don't know why 😕 , given that the API key should [behave similar to an OAuth token](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/).

Please see more discussion in the Epic issue 2613: `common-samples/issues/2613`

Reply:
Also @TADraeseke to chime in here.

Reply:
We didn't clear the key like this on Android or Java--it shouldn't be necessary unless there's something I'm missing.

Reply:
Discussed this during Friday review, and it seems Android/Java have different Sample Viewer architecture. To my understanding Android doesn't set an environment key as each sample is sort of standalone, thus they don't clear the key in init. Meanwhile, dotnet does clear the key https://github.com/Esri/arcgis-runtime-samples-dotnet/pull/840#discussion_r562973021 on init, and reload it on destroy.

Reply:
Thank you @yo1995 and @TADraeseke; that makes sense. However, since this change is only needed when saving the map to the portal, could it be done before the call to `save(as:portal:tags:folder:itemDescription:thumbnail:forceSaveToSupportedVersion:completion:)` and undone in the completion handler? That way no `apiKey` property is needed and `viewDidLoad()` isn't doing hokey stuff. While quite common, it generally isn't a good idea to perform actions in `viewDidLoad()` and undo them in `deinit`. For one thing, `viewDidLoad()` isn't guaranteed to be called. It is possible (though rare) for a view controller to be instantiated but never presented, in which case deinit would be called without `viewDidLoad()` being called. Also, in most cases, the work being performed in `viewDidLoad()` has nothing to do with the view and therefore would be better off in the initializer itself.

Reply:
> could it be done before the call to `save...` and undone in the completion handler?

Actually, can I even recover the key immediately after `save` is called (https://github.com/Esri/arcgis-runtime-samples-ios/pull/1067/commits/f371652d928b28c20aa1d94e98031ef1870a557b), as there is a chance that it never calls back due to connection issue or so? Also, please see my comment below.

> ... deinit would be called without viewDidLoad() being called.

Yeah that is my fear too 🤕